ui/pages/confirmations/hooks/alerts/signatures/useAccountMismatchAlerts.ts

```diff
@@ -0,0 +1,54 @@
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import { Alert } from '../../../../../ducks/confirm-alerts/confirm-alerts';
+import { Severity } from '../../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+import { currentConfirmationSelector } from '../../../../../selectors';
+import { getConfirmationSender } from '../../../components/confirm/utils';
+import { SignatureRequestType } from '../../../types/confirm';
+import { isSIWESignatureRequest } from '../../../utils';
+
+/**
+ * This hook returns an array of alerts when the expected address of the request
+ * does not match the selected account's address.
+ */
+export default function useAccountMismatchAlerts(): Alert[] {
+  const t = useI18nContext();
+
+  const currentConfirmation = useSelector(
+    currentConfirmationSelector,
+  ) as SignatureRequestType;
+  const { from: fromAddress } = getConfirmationSender(currentConfirmation);
+
+  const isSIWE = isSIWESignatureRequest(currentConfirmation);
+  const siweParsedAddress =
+    currentConfirmation?.msgParams?.siwe?.parsedMessage?.address;
+  const isMismatchSIWEAdddress =
+    siweParsedAddress?.toLowerCase() !== fromAddress?.toLowerCase();
+  const isMismatchAccount = isSIWE && isMismatchSIWEAdddress;
+
+  const alerts = useMemo(() => {
+    if (!isMismatchAccount) {
+      return [];
+    }
+
+    return [
+      {
+        field: 'account',
+        key: 'account',
+        message: t('alertMessageSignInWrongAccount'),
+        reason: t('alertReasonWrongAccount'),
+        severity: Severity.Warning,
+      },
+      {
+        field: 'signingInWith',
+        key: 'signingInWith',
+        message: t('alertMessageSignInWrongAccount'),
+        reason: t('alertReasonWrongAccount'),
+        severity: Severity.Warning,
+      },
+    ] as Alert[];
+  }, [isMismatchAccount, t]);
+
+  return alerts;
```

Comment:
Minor, but can we `return` the hook directly to make it explicit we're caching the result?

Reply:
updated https://github.com/MetaMask/metamask-extension/pull/25613